src/query/split/split_trimesh.rs

```diff
@@ -344,6 +346,274 @@ impl TriMesh {
+    /// equal to `normal * bias`).
+    pub fn canonical_intersection_with_plane(
+        &self,
+        axis: usize,
+        bias: Real,
+        epsilon: Real,
+    ) -> IntersectResult<Vec<Polyline>> {
+        self.intersection_with_local_plane(&Vector::ith_axis(axis), bias, epsilon)
+    }
+
+    /// Computes the intersection [`Polyline`]s between this mesh, transformed by `position`,
+    /// and a plane identified by its normal `axis` and the `bias`
+    /// (i.e. the plane passes through the point equal to `normal * bias`).
+    pub fn intersection_with_plane(
+        &self,
+        position: &Isometry<Real>,
+        axis: &UnitVector<Real>,
+        bias: Real,
+        epsilon: Real,
+    ) -> IntersectResult<Vec<Polyline>> {
+        let local_axis = position.inverse_transform_unit_vector(axis);
+        let added_bias = -position.translation.vector.dot(&axis);
+        self.intersection_with_local_plane(&local_axis, bias + added_bias, epsilon)
+    }
+
+    /// Computes the intersection [`Polyline`]s between this mesh
+    /// and a plane identified by its normal `local_axis`
+    /// and the `bias` (i.e. the plane passes through the point equal to `normal * bias`).
+    pub fn intersection_with_local_plane(
+        &self,
+        local_axis: &UnitVector<Real>,
+        bias: Real,
+        epsilon: Real,
+    ) -> IntersectResult<Vec<Polyline>> {
+        // 1. Partition the vertices.
+        let vertices = self.vertices();
+        let indices = self.indices();
+        let mut colors = vec![0u8; self.vertices().len()];
+
+        // Color 0 = on plane.
+        //       1 = on negative half-space.
+        //       2 = on positive half-space.
+        let mut found_negative = false;
+        let mut found_positive = false;
+        for (i, pt) in vertices.iter().enumerate() {
+            let dist_to_plane = pt.coords.dot(&local_axis) - bias;
+            if dist_to_plane < -epsilon {
+                found_negative = true;
+                colors[i] = 1;
+            } else if dist_to_plane > epsilon {
+                found_positive = true;
+                colors[i] = 2;
+            }
+        }
+
+        // Exit early if `self` isn’t crossed by the plane.
+        if !found_negative {
+            return IntersectResult::Positive;
+        }
+
+        if !found_positive {
+            return IntersectResult::Negative;
+        }
+
+        // 2. Split the triangles.
+        let mut index_adjacencies = HashMap::default(); // Adjacency list of indices
+
+        // Helper function for adding polyline segments to the adjacency list
+        let mut add_segment_adjacencies = |idx_a, idx_b| {
+            let _ = index_adjacencies
+                .entry(idx_a)
+                .and_modify(|list: &mut HashSet<u32>| {
+                    let _ = list.insert(idx_b);
+                })
+                .or_insert_with(|| {
+                    let mut adjacencies = HashSet::with_capacity(2);
```

Comment:
I noticed that the `HashMap`s used in other functions are re-exports of `FxHashMap`. I wasn't sure if I should use the fx variant of HashSet, so I just used the `std` version. I'm happy to change this if it's an issue.

Reply:
I had it working like this originally, but it's possible for the same neighbor to be added twice if multiple, consecutive triangle segments lie in the intersection plane. That being said, it shouldn't actually cause any issues to have those duplicates (other than a very minor perf hit). I'll rework :)

Reply:
Fixed as of 6d2cdc4b7dcffd945a86fb2b908a31e8200fe029